File: app/src/main/java/com/easy/kotlins/http/core/OkDownloadCallback.java
package com.easy.kotlins.http.core;

import java.io.File;

/**
 * Create by LiZhanPing on 2020/7/18
 * desc: 下载监听
 */
public interface OkDownloadCallback extends OkStartedCallback<File>{

    void onProgress(long downloadedBytes, long totalBytes);

    void onCancel();
}

File: common-ktx/src/main/java/com/easy/kotlins/http/core/OkDownloadCallback.kt
package com.easy.kotlins.http.core

import java.io.File

/**
 * Create by LiZhanPing on 2020/7/18
 * desc: 下载监听
 */
interface OkDownloadCallback : OkCallback<File> {
    fun onProgress(downloadedBytes: Long, totalBytes: Long)
    fun onCancel()
}